src/server.ts

```diff
@@ -260,25 +260,32 @@ function slicePayload(payload, batchSize, ecosystem): any {
 const regexVersion =  new RegExp(/^([a-zA-Z0-9]+\.)?([a-zA-Z0-9]+\.)?([a-zA-Z0-9]+\.)?([a-zA-Z0-9]+)$/);
 const sendDiagnostics = async (ecosystem: string, diagnosticFilePath: string, contents: string, collector: IDependencyCollector) => {
     connection.sendNotification('caNotification', {'data': caDefaultMsg});
-    const deps = await collector.collect(contents);
-    let validPackages = deps
-    if (ecosystem != "golang") {
-        validPackages = deps.filter(d => regexVersion.test(d.version.value.trim()));
+    let deps = null;
+    try {
+        deps = await collector.collect(contents, diagnosticFilePath);    
+
+        let validPackages = deps
+        if (ecosystem != "golang") {
+            validPackages = deps.filter(d => regexVersion.test(d.version.value.trim()));
+        }
+        const requestPayload = validPackages.map(d => ({ "package": d.name.value, "version": d.version.value }));
+        const requestMapper = new Map(validPackages.map(d => [d.name.value + d.version.value, d]));
+        const batchSize = 10;
+        let diagnostics = [];
+        let totalCount = new TotalCount();
+        const start = new Date().getTime();
+        const allRequests = slicePayload(requestPayload, batchSize, ecosystem).map(request =>
+            fetchVulnerabilities(request).then(response =>
+                runPipeline(response, diagnostics, diagnosticFilePath, requestMapper, totalCount)));
+    
+        await Promise.allSettled(allRequests);
+        const end = new Date().getTime();
+        console.log("Time taken to fetch vulnerabilities: " + ((end - start) / 1000).toFixed(1) + " sec.");
+        connection.sendNotification('caNotification', { 'data': getCAmsg(deps, diagnostics, totalCount), 'diagCount': diagnostics.length > 0 ? diagnostics.length : 0 });
+    } catch (error) {
+        console.error("Command execution failed, something wrong with manifest file go.mod\n%s", error)
+        connection.sendNotification('caError', {'data': 'Invalid menifest file or need to syncronize package details.'})
     }

```

Comment:
Add try block only to the statement which could potentially raise an exception. Don't include all statements.

Reply:
collect.collect() raises exception, but upon exception I don't want to execute rest of the steps. So kept the entire block under try-catch block.

Reply:
In that case you can return from the exception block rather than putting everything under try.

Reply:
This is done